spring-integration-jms/src/main/java/org/springframework/integration/jms/SubscribableJmsChannel.java

```diff
@@ -238,7 +240,9 @@ else if (this.logger.isWarnEnabled()) {
 					}
 				}
 				else {
-					throw new MessageDeliveryException(messageToSend, exceptionMessage, e);
+					throw messageToSend != null
+							? new MessageDeliveryException(messageToSend, exceptionMessage, e)
+							: new MessageDeliveryException(exceptionMessage);
```

Comment:
This is not correct.
We are in a `MessageDispatchingException` case, which can happen only from the `this.dispatcher.dispatch(messageToSend);`.
Therefore `messageToSend` is never null here when we reach this place.
So, I suggest to have a `@SuppressWarnings("NullAway")` as we discussed before.